Lombiq.HelpfulLibraries/Libraries/Database/SchemaBuilderExtensions.cs

```diff
@@ -16,5 +18,31 @@ public static class SchemaBuilderExtensions
                 .CreateIndex(
                     $"IDX_{typeof(T).Name}_{DocumentId}",
                     DocumentId));
+
+        /// <summary>
+        /// Alters the table with the same name as the typename of <typeparamref name="TTable"/> to add a database index
+        /// for the given columns with a <c>IDX_tableName_columnNames</c> naming scheme.
+        /// </summary>
+        /// <param name="schemaBuilder">The schema builder.</param>
+        /// <param name="columnNames">The collection of columns the index should apply to.</param>
+        /// <typeparam name="TTable">
+        /// The type with the same name as the database table (e.g. name of an index type).
+        /// </typeparam>
+        /// <returns>The <paramref name="schemaBuilder"/> so it can be chained.</returns>
+        /// <exception cref="ArgumentException">
+        /// It is thrown when the <paramref name="columnNames"/> is null or empty.
+        /// </exception>
+        public static ISchemaBuilder AddDatabaseIndex<TTable>(this ISchemaBuilder schemaBuilder, params string[] columnNames)
+        {
+            if (columnNames?.Any() != true)

```

Comment:
I think if you have empty or null column names that's already wrong. We should not hide that mistake because the consumer must've messed something up bad.

```csharp
            if (columnNames.Any(string.IsNullOrWhiteSpace))
            {
                throw new ArgumentException(
                    "The column names shouldn't be null, empty string or all whitespace.",
                    nameof(columnNames));
            }
```

Reply:
Can't agree more!